src/annotation/app.py

```diff
@@ -0,0 +1,147 @@
+    columns = [column[0] for column in cur.description]
+    results = []
+    for row in cur.fetchall():
+        row = dict(zip(columns, row))
+        row['geometry'] = json.loads(row['geometry'])
+        row['metadata'] = json.loads(row['metadata'])
+        results.append(row)
+
+    return jsonify(results)
+
+
+@app.route("/post_annotation", methods=['POST'])
+def post_annotation():
+    req = request.get_json()
+    logging.info("POST request: {}".format(req))
+
+    # TODO: validate the payload.
+    # TODO: escape quotes and other dangerous chars
+    con = get_db()
+    cur = con.cursor()
+    # anno_id , img_id , geometry , annotation , metadata 
+    cmd = "INSERT INTO annotations (anno_id,img_id,geometry,annotation,metadata) VALUES (null, {}, '{}', '{}', '{}')".format(
+        req['img_id'], json.dumps(req['geometry']), req['annotation'], json.dumps(req['metadata'])
+    )
+    logging.info(cmd)
+    cur.execute(cmd)
+    con.commit()
+    cur.execute('SELECT COUNT(*) AS c FROM annotations')
+    result = next(cur, [None])['c']
+    logging.info('found {} rows'.format(result))
+    return jsonify({"message": "Annotation posted", "id": cur.lastrowid})
+
+
+@app.route("/replace_annotation/<int:anno_id>", methods=['POST'])
+def replace_annotation(anno_id):
+    req = request.get_json()
+    logging.info("POST replace request: {}".format(req))
+
+    # TODO: validate the payload.
+    # TODO: escape quotes and other dangerous chars
+    # TODO: change this to an update rather than delete/insert?
+    con = get_db()
+    cur = con.cursor()
+    # anno_id , img_id , geometry , annotation , metadata 
+    cmd1 = "DELETE FROM annotations WHERE anno_id={}".format(anno_id)
+    cmd = "INSERT INTO annotations (anno_id,img_id,geometry,annotation,metadata) VALUES (null, {}, '{}', '{}', '{}')".format(
+        req['img_id'], json.dumps(req['geometry']), req['annotation'], json.dumps(req['metadata'])
+    )
+    logging.info(cmd1)
+    logging.info(cmd)
+    cur.execute(cmd1)
+    cur.execute(cmd)
+    con.commit()
+    cur.execute('SELECT COUNT(*) AS c FROM annotations')
+    result = next(cur, [None])['c']
+    logging.info('found {} rows'.format(result))
+    return jsonify({"message": "Annotation posted", "id": cur.lastrowid})
+
+
+@app.route("/delete_annotation/<int:anno_id>", methods=['DELETE'])
+def delete_annotation(anno_id):
+    req = request.get_json()
+    logging.info("DELETE request: {}".format(req))
+
+    # TODO: validate the payload.    
+    con = get_db()
+    cur = con.cursor()
+    cmd = "DELETE FROM annotations WHERE anno_id={}".format(anno_id)
+    logging.info(cmd)
+    cur.execute(cmd)
+    con.commit()
+    return jsonify(success=True)
+
+
+@app.route("/annotate/")
+@app.route("/annotate/<int:id>")
+def annotate(id=None):
+    show_annot = request.args.get('show_annot')
+    if id is None:
+        cur = get_db().cursor()
+        if show_annot:
+            cur.execute("SELECT MIN(images.img_id) AS id from images")
```

Comment:
bad things happen if the number of images is 0. That's probably not a big problem.